### DISCUSSION TITLE
OSQuery Windows UTF8 Errors

### DISCUSSION THREAD

**Discussion by B3DTech** [Security-Onion-Solutions/securityonion#2675] [Unsupported Versions]
SO 2.3.21 distributed install

With the latest release of OSQuery (4.5.1) as packaged and available on the downloads page in SOC installed on my Windows Hosts, I'm still seeing the following errors flooding my Application Event Logs:

caller=log.go:124 ts=2021-01-19T16:52:01.0215442Z caller=level.go:63 level=info caller=log.go:69 component=osquery level=stderr msg="I0119 11:52:01.021148 11232 registry.cpp:558] Failed to expand globs: Failed to open registry handle" caller=registry.cpp:558

caller=log.go:124 ts=2021-01-19T16:50:26.1264442Z caller=level.go:63 level=info caller=publish_logs.go:179 method=PublishLogs uuid=6e2ccedd-1f13-4ea9-9bcc-37d9611e3eff logType=string log_count=1103 message= errcode= reauth=false err="rpc error: code = Internal desc = grpc: error while marshaling: proto: field \"kolide.agent.LogCollection.Log.Data\" contains invalid UTF-8" took=14.0018ms

Additionally seeing the following:
caller=log.go:124 ts=2021-01-19T17:22:49.407223Z caller=level.go:63 level=info caller=log.go:69 component=osquery level=stderr msg="I0119 12:22:49.407222 15600 registry.cpp:558] Failed to expand globs: Failed to open registry handle" caller=registry.cpp:558

https://github.com/osquery/osquery/pull/6338 was supposed to have resolved the issue, which was merged a while ago.

**Comment by defensivedepth** [comment]
@B3DTech Can you confirm the installed version number? `SELECT version FROM osquery_info;`

Also - are you getting any Windows Eventlogs through osquery?

**Reply by B3DTech** [reply]
4.5.1, as verified through the query and the properties of the executable.

select * from windows_events yields no records for the machine with 4.5.1 on it.

For machines with 4.2.0, I do get Windows event logs, and they also contain the same error messages regarding UTF-8

**Reply by defensivedepth** [reply]
Ok so to confirm - on the system that has 4.5.1 installed, you are getting these errors in the Application eventlog (which you originally posted):

----------

caller=log.go:124 ts=2021-01-19T16:52:01.0215442Z caller=level.go:63 level=info caller=log.go:69 component=osquery level=stderr msg="I0119 11:52:01.021148 11232 registry.cpp:558] Failed to expand globs: Failed to open registry handle" caller=registry.cpp:558

caller=log.go:124 ts=2021-01-19T16:50:26.1264442Z caller=level.go:63 level=info caller=publish_logs.go:179 method=PublishLogs uuid=6e2ccedd-1f13-4ea9-9bcc-37d9611e3eff logType=string log_count=1103 message= errcode= reauth=false err="rpc error: code = Internal desc = grpc: error while marshaling: proto: field "kolide.agent.LogCollection.Log.Data" contains invalid UTF-8" took=14.0018ms

Additionally seeing the following:
caller=log.go:124 ts=2021-01-19T17:22:49.407223Z caller=level.go:63 level=info caller=log.go:69 component=osquery level=stderr msg="I0119 12:22:49.407222 15600 registry.cpp:558] Failed to expand globs: Failed to open registry handle" caller=registry.cpp:558

**Reply by B3DTech** [reply]
I am receiving the errors on all Windows clients, both 4.2.0 and 4.5.1

caller=level.go:63 level=info caller=extension.go:494 err="sending string logs: writing logs: transport error sending logs: rpc error: code = Internal desc = grpc: error while marshaling: proto: field \"kolide.agent.LogCollection.Log.Data\" contains invalid UTF-8"

caller=level.go:63 level=info caller=publish_logs.go:157 method=PublishLogs uuid=c6c658be-5cb1-4d4d-b6d6-5e0f61e14de9 logType=string log_count=754 message= errcode= reauth=false err="rpc error: code = Internal desc = grpc: error while marshaling: proto: field \"kolide.agent.LogCollection.Log.Data\" contains invalid UTF-8" took=3.0003ms

**Reply by defensivedepth** [reply]
I am chatting about this over in the osquery Slack; I see you have posted on the osquery github issue as well.

I'm trying to replicate this, but not able to; what kind of logs are you shipping? Just regular System/Security/Application? Sysmon logs as well?

**Reply by B3DTech** [reply]
Yup - you got it.  System,Application,Security,Microsoft-Windows-Sysmon/Operational

The folks at Launcher say that while Osquery is creating the non-utf8 data, Launcher is throwing the errors, so they are going to try to heal the data and if It can't, redact it.  Not sure that's a good option.  I don't want missing data, especially security related data.  I don't want Osquery to create the non-utf8 data in the first place.

https://github.com/kolide/launcher/issues/445

**Reply by defensivedepth** [reply]
I still haven't been able to replicate the issue. What language set is the system running? What type of logs are you generating with sysmon?

**Reply by B3DTech** [reply]
Hey Josh - our systems are US English.  The sysmon is using a config I saw highly recommended from Swift On Security.  I don't believe I did too much if any out of the box config of the XML.  https://github.com/SwiftOnSecurity/sysmon-config/blob/master/sysmonconfig-export.xml

**Reply by defensivedepth** [reply]
Does this happen on a fresh W10 / Osquery install? 

I'm wondering if there is mis-formatted osquery results in the osquery store from the previous version of osquery, and Launcher is still trying to send that. You can either try a fresh install with 4.5.1 or blow away C:\Program Files\Kolide\Launcher-so-launcher\data\osquery.db and restart the service (that should remove all the previous results).

**Reply by B3DTech** [reply]
Freaking nailed it.  Deleted the data folder and let it recreate it on service restart - no UTF8 errors.  OSQuery is still pretty chatty at start, but no continual UTF8 events.  Thank you!

**Reply by defensivedepth** [reply]
Fantastic! Would you mind noting that in the osquery issue?
